__tests__/src/middleware/auth.test.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+
+const app = require('../../../src/app.js');
+const superagent = require('superagent');
+
+
+
+describe('api/signup', () => {
+
+  const PORT = 8888;
+  beforeAll(() => {
+    app.start(PORT);
+  });
+  afterAll(() => app.stop());
+
+
+  it('Throws a 400 error on a bad model', () => {
+    return superagent.post('http://localhost:8888/signup')
+      .send({
+        username: 'somethingsomething',
+        password: 'unknown',
+      })
+      .then(Promise.reject)
+      .catch(response => {
+        console.log('hello', response.status);
+        expect(response.status).toEqual(400);
+      });
+  });
+
```

Comment:
This test looks good - make sure you have the following:

create a test that will ensure that your API returns a status code of 404 for any routes that have not been registered
/api/signup
POST - test 400, if no request body has been provided or the body is invalid
POST - test 200, if the request body has been provided and is valid
/api/signin
GET - test 401, if the user could not be authenticated
GET - test 200, responds with token for a request with a valid basic authorization header